app/components/agenda/agenda-header/agenda-version-actions.js

```diff
@@ -342,22 +342,25 @@ export default class AgendaAgendaHeaderAgendaVersionActions extends Component {
         undefined,
         this.intl.t('an-agenda-was-approved-since-modal-was-opened')
       );
-      const agendaitemsNotOk = await this.allAgendaitemsNotOk();
       const newAgendaId = await approveDesignAgenda(this.args.currentAgenda);
       const newAgenda = await this.store.findRecord('agenda', newAgendaId);
+      const agendaCheckMappingsMap = new Map(
+        this.agendaCheckMapping.map(({ uri, generatedName }) => [uri, generatedName])
+      );
       await this.documentService.setGeneratedPieceNames(
         this.args.currentAgenda.id,
-        this.agendaCheckMapping,
+        agendaCheckMappingsMap,
         this.openAgendaCheckTimestamp,
       );
       await this.documentService.stampDocumentsOfAgenda(this.args.currentAgenda.id);
       // Data reloading
       await this.reloadAgenda(this.args.currentAgenda);
       await this.reloadAgendaitemsOfAgenda(this.args.currentAgenda);
       await this.reloadMeeting();
-      if (agendaitemsNotOk?.length) {
-        await this.decisionReportGeneration.regenerateDecisionReportsForMeeting.perform(this.args.meeting, true);
-      }
+      // Regenerate ALL beslissingsfiches, so we ensure that the concerns part
+      const newNames = true;
+      const agendaitemsToRegenerateConcernFor = [...new Set(this.agendaCheckMapping.map(({ agendaitem }) => agendaitem.id))];
+      await this.decisionReportGeneration.regenerateDecisionReportsForMeeting.perform(this.args.meeting, newNames, agendaitemsToRegenerateConcernFor);
```

Comment:
this.decisionReportGeneration.regenerateDecisionReportsForMeeting only when there are mappings. or we will regen everything when there are no changes